### ISSUE TITLE
Naming inconsistencies

### ISSUE THREAD

**Issue by dmitriy-serdyuk** [pytorch/pytorch#1969]
There are some functions/classes which are very similar but have a different naming pattern. For example:
- `log_softmax` vs `logsigmoid`: underscore symbol
- `CrossEntropyLoss` and `NLLLoss` vs `BCELoss` and `BCEWithLogitsLoss`: CE loss and BCE loss have different inputs

Maybe there are other examples. Is it the legacy of lua torch?

**Comment by fmassa** [comment]
Good catch for the `log_softmax`/`logsigmoid`, maybe we should deprecate one naming in favor of the other?
About `CrossEntropyLoss` and `NLLLoss`, if I remember properly it was intentional to keep `CrossEntropy` because `CE` is not as widespread as `NLL`. `BCEWithLogitsLoss` is very new and could (eventually?) be renamed.

**Comment by dmitriy-serdyuk** [comment]
From my point of view `log_sigmoid` would look better, more readable.

I meant that the inputs are different for those losses, not than one abbreviates the name and other one doesn't. I think, it makes sense to rename `BCEWithLogitsLoss` into `BCELoss` and `BCELoss` into something like `BinaryNLLLoss` so that its inputs types correspond to the multiclass analogs.

**Comment by alykhantejani** [comment]
I think the issue with renaming `BCEWithLogitsLoss` to `BCELoss` is that is no longer backwards compatible. i.e. people currently using `BCELoss` pass the input through a sigmoid first which will break with `BCEWithLogitsLoss`

**Comment by zou3519** [comment]
It's probably good to fix the inconsistencies. I know that the CrossEntropyLoss vs BCELoss difference confuses  many of our users
